### PR TITLE
Various updates, cleanup

### PR DESCRIPTION
**Pull request by process**
Earlier this year I made a few simple cleanups to the code but never bothered to make a pull request. They should merge fine with the google analytics stuff I think.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+The de'Barbari Explorer
+=======================
+
+This is the source code used to power a interactive, explorable,
+annotatable map of the Venice as it appeared in the year 1500.
+
```
Comment (bennlich) [inline README.md:6]: Should probably read:

This is the source code used to power **an** interactive, explorable, annotatable map of ~~the~~ Venice as it appeared in the year 1500.

File: debarbari.js [hunk 2]
```diff
@@ -1,69 +1,122 @@
...
-    if (list[i].properties && list[i].properties.name == val) return list[i];
-  }
+// Turn on strict mode
+"use strict";
 
-  return null;
-}
+var map,
+    DATA = [],
+    Firebase,
```
Comment (bennlich) [inline debarbari.js]: Hmm, why is `Firebase` declared here? Doesn't it already exist in the global scope?

File: debarbari.js [hunk 3]
```diff
@@ -1,69 +1,122 @@
...
+        if (datasets[i][j].properties && datasets[i][j].properties.name) {
+          names.push(datasets[i][j].properties.name);
+        }
+      }
+    }
+
+    return names;
+  },
+
+  /* GetJSON uses x/y coordinates, whereas
```
Comment (bennlich) [inline debarbari.js]: Is "GetJSON" supposed to be "GeoJSON"? Also, this comment seems to imply some kind of coordinate conversion, but it looks like its purpose is more to restructure a list of data. Is that right?

File: debarbari.js [hunk 4]
```diff
@@ -1,69 +1,122 @@
...
     // Register handlers
-    $("#dlbutton").click(function () { 
+    $("#dlbutton").click(function () {
         var link = document.createElement("a");
-        link.href = getData();
+        link.href = downloader.getData();
         link.download = "debarbari.png";
         var theEvent = document.createEvent("MouseEvent");
-        theEvent.initMouseEvent("click", true, true, window, 0, 0, 0, 0, 0, false, false, false, false, 0, null);
+        theEvent.initMouseEvent("click", true, true, window, 0, 0, 0, 0, 0,
```
Comment (bennlich) [inline debarbari.js:114]: Could be good to include a comment here about why you're creating a MouseEvent. Is it to get around the popup blocker?

**Comment by bennlich** [comment]
Awesome changes overall!

File: debarbari.js (hunk 3 above)
Comment (process) [inline debarbari.js]: Yep, sorry for the typo. When we originally wrote that we probably thought we were going to try to actually convert pixel points to lat/lng, but it became clear that wouldn't be feasible.

Comment (bennlich) [inline debarbari.js]: Nothing to be sorry for! Just thought I'd point out things I noticed. Not sure if you know this--if you push changes to the branch that this PR is based on, the PR will use those latest changes. So if you think any of my comments are change-worthy, you can push them to the branch, and then we can merge.

**Comment by fabio-carrera** [comment]
Dear Justin and Benny:

thanks for all the work you've done on this wonderful application.
This year, a WPI Venice team is going to try to use this same approach to
annotate OTHER maps of Venice.
Ideally, we will try to merge this application with the EVOLUTION
http://evolution.veniceprojectcenter.org/evolution.html app (with a
redesigned full-screen interface with the slider on top of the image).

I am telling you this just as a heads up, since we may want to fork off the
NEW generic map annotation map, before we start the process of generalizing
it.

In the new generic app, I want the LAYERS to be shared, so that the user
can SELECT -- say -- a church from a single overall list (even if created
on another map), so that we can then track the SAME church over all the
maps.

Anyway, the new WPI students may be capable of making the changes on their
own, with guidance from you both.

_CHANGES TO THE CURRENT VERSION_
In the meantime, there is one feature that would be very useful right now,
in addition to addressing Kyle's issues:
https://docs.google.com/document/d/1WjsHJ3AMPm1LOnlK9mFT7M4xZ7prDkvcReBcKTdXWJM/edit#
- Add a Delete and an Edit button within the Bubble, where it already
  says _clone_ on the left,
- Allow these 3 functions (clone, edit, and delete) only if one is
  logged in (of course)
  - EDIT will allow the editing of the label, which currently only
    shows up at the end of the editing, and may allow editing of the
    polygon's
    colors (with option to use layer default)
  - DELETE will ask confirmation
- Allow the editing of the COLORs for a whole Layer
- Create a single index of names within each layer so that other old
  maps can refer to the same item (church, island)

Thanks again

Fabio

Fabio Carrera carrera.fabio@gmail.com, Ph.D.
Cell.:  +1 508-615-5333 (USA) | +39 335 581-5292 (Italia)
Skype: carrerawpi | Twitter http://twitter.com/fabiocarrera | Facebook
http://www.facebook.com/#!/profile.php?id=29105207 | Blog
http://venice2point0.blogspot.com/ | Homepage http://fabiocarrera.com/
Dashboard http://dashboard.veniceprojectcenter.org | Venice 25th
http://veniceprojectcenter.org | Venipedia http://www.venipedia.org |
VPC https://sites.google.com/site/vpcprep/ | SFPC
https://sites.google.com/site/santafeprojectcenter/

File: debarbari.js (hunk 3 above)
Comment (process) [inline debarbari.js]: Yup, I know. Since these changes are so small, I'll update the pull request soon.

**Comment by process** [comment]
I fixed the typo and other simple mistakes

Fabio:
:+1: Sounds great! I'm meeting with that team soon, and will be happy to assist them as needed.

**Pull request merged**